irksome/stepper.py

```diff
@@ -70,16 +70,10 @@ def _update(self):
         ns = self.num_stages
         nf = self.num_fields
 
-        if nf == 1:
-            ks = self.ks
-            for i in range(ns):
-                u0.dat.data[:] += dtc * b[i] * ks[i].dat.data[:]
-        else:
-            k = self.stages
-
-            for s in range(ns):
-                for i in range(nf):
-                    u0.dat.data[i][:] += dtc * b[s] * k.dat.data[nf*s+i][:]
+        ks = self.ks
+        for s in range(ns):
+            for i, u0d in enumerate(u0.dat):
```

Comment:
A dat, even if not mixed, is iterable (yielding itself in the non-mixed case) so one doesn't have to special case mixed vs. non-mixed in the stage application. Since the `ks` are also only read, I access with `.data_ro` which will be slightly cheaper.

Reply:
Didn't realize this.  The also simplifies the logic going forward.